pkgs/games/vcmi/default.nix

```diff
@@ -0,0 +1,84 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, SDL2
+, SDL2_image
+, SDL2_mixer
+, SDL2_ttf
+, autoPatchelfHook
+, boost
+, cmake
+, ffmpeg
+, innoextract
+, luajit
+, ninja
+, pkg-config
+, qtbase
+, tbb
+, wrapQtAppsHook
+, zlib
+, testers
+, vcmi
+}:
+
+stdenv.mkDerivation rec {
+  pname = "vcmi";
+  version = "unstable-2022-09-11";
+
+  src = fetchFromGitHub {
+    owner = "vcmi";
+    repo = "vcmi";
+    rev = "cedc9a92ede66b3f9fff079022ea287a012330a5";
+    fetchSubmodules = true;
+    hash = "sha256-5PuFq6wDSj5Ye2fUjqcr/VRU0ocus6h2nn+myQTOrhU=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    cmake
+    ninja
+    pkg-config
+    wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    SDL2
+    SDL2_image
+    SDL2_mixer
+    SDL2_ttf
+    boost
+    ffmpeg
+    luajit
+    qtbase
+    tbb
+    zlib
+  ];
+
+  postPatch = ''
+    substituteInPlace Version.cpp.in \
+      --subst-var-by GIT_SHA1 "${src.rev}";
```

Comment:
When I get past main menu and run actual scenario the program fails to load it's libraries:

```
Cannot open dynamic library (/nix/store/gnjm0qjp69bndm9dpilmc0zk8kgkcrmg-vcmi-unstable-2022-09-11//nix/store/gnjm0qjp69bndm9dpilmc0zk8kgkcrmg-vcmi-unstable-2022-09-11/lib/vcmi/AI/libVCAI.so). Throwing...
terminate called after throwing an instance of 'std::runtime_error'
  what():  Cannot open dynamic library
```

Locally I have been working it around with

```
  cmakeFlags = [
    # Make libvcmi.so discoverable in non-standard location.
    # Default rpath does not contain it at all.
    "-DCMAKE_INSTALL_RPATH=${placeholder "out"}/lib/vcmi"

    "-DENABLE_PCH=OFF"

    # Upstream assumes relative value whilc nixpkgs passes absolute.
    # Both should be allowed:
    #  https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html
    # Meawhile work it around by passing relative value.
    "-DCMAKE_INSTALL_LIBDIR=lib"
  ];
```

Reply:
Thanks, I've added these.